src/toil/batchSystems/mesos/executor.py

```diff
@@ -212,11 +233,75 @@ def frameworkMessage(self, driver, message):
         log.debug("Received message from framework: {}".format(message))
 
 
-def main(executorClass=MesosExecutor):
+def main():
     logging.basicConfig(level=logging.DEBUG)
     log.debug("Starting executor")
-    executor = executorClass()
-    driver = mesos.native.MesosExecutorDriver(executor)
-    exit_value = 0 if driver.run() == mesos_pb2.DRIVER_STOPPED else 1
+
+    if not os.environ.has_key("MESOS_AGENT_ENDPOINT"):
+        # Some Mesos setups in our tests somehow lack this variable. Provide a
```

Comment:
That does seem strange.  Did you get any further in solving this one?

Reply:
Not really. I'm not sure it happens in 1.01, either. I added this before we switched.